modules/uhttp/client.go

```diff
@@ -0,0 +1,100 @@
+package uhttp
+
+import (
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"golang.org/x/net/context/ctxhttp"
+
+	"go.uber.org/fx/core"
+)
+
+// Client wraps around a http client
+type Client struct {
+	http.Client
+	filters []ClientFilter
+}
+
+// NewClient creates a new instance of uhttp Client
+func NewClient(client http.Client, filters ...ClientFilter) *Client {
+	if filters == nil {
+		filters = []ClientFilter{ClientFilterFunc(tracingClientFilter)}
+	}
+	return &Client{Client: client, filters: filters}
```

Comment:
Yep that makes sense. Clients can pass custom filters but they wouldn't be passing the tracing filter. Any suggestions on how I can restrict struct initialization to just the new method? I basically have duplicate filter checks in NewClient and Do methods since they can initialize the struct method directly. I cannot have the exported NewClient method return an unexported client struct. I cannot make an interface either since the http.Client is a struct.